app/controllers/register/VariationWorkingKnowledgeController.scala

```diff
@@ -67,16 +66,26 @@ class VariationWorkingKnowledgeController @Inject()(
             case None => true
             case Some(existing) => existing != value
           }
-          cacheConnector.save(request.externalId, IsNewAdviserId, !value).flatMap(_ =>
-            if (hasAnswerChanged) {
-              cacheConnector.save(request.externalId, VariationWorkingKnowledgeId, value).flatMap(cacheMap =>
-                saveChangeFlag(mode, VariationWorkingKnowledgeId).map(_ =>
+
+          val resultOfSaveDeclarationFlag = mode match {
+            case CheckUpdateMode =>
+              cacheConnector.save (request.externalId, PAInDeclarationJourneyId, true)
+            case _ =>
+              Future.successful(())
+          }
+
+          resultOfSaveDeclarationFlag.flatMap(_ =>
+            cacheConnector.save(request.externalId, IsNewAdviserId, !value).flatMap(_ =>
+              if (hasAnswerChanged) {
```

Comment:
can we removed hasAnswerChanged conditions? as it was decided we are not going to compare data

Reply:
done